examples/messaging/oracle-aq-websocket-mp/src/main/java/io/helidon/examples/messaging/mp/MsgProcessingBean.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c)  2020 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.helidon.examples.messaging.mp;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.SubmissionPublisher;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.jms.JMSException;
+import javax.jms.MapMessage;
+
+import io.helidon.common.reactive.BufferedEmittingPublisher;
+import io.helidon.common.reactive.Multi;
+import io.helidon.messaging.connectors.aq.AqMessage;
+
+import org.eclipse.microprofile.reactive.messaging.Acknowledgment;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.reactivestreams.FlowAdapters;
+import org.reactivestreams.Publisher;
+
+@ApplicationScoped
+public class MsgProcessingBean {
+
+    private final BufferedEmittingPublisher<String> emitter = BufferedEmittingPublisher.create();
+    private final SubmissionPublisher<String> broadCaster = new SubmissionPublisher<>();
+
+    @Outgoing("to-queue-1")
+    public Publisher<String> toFirstQueue() {
+        // Create new publisher for emitting to by this::process
+        return ReactiveStreams
+                .fromPublisher(FlowAdapters.toPublisher(emitter))
+                .buildRs();
+    }
+
+    @Incoming("from-queue-1")
+    @Outgoing("to-queue-2")
+    //Leave commit by ack to outgoing connector
+    @Acknowledgment(Acknowledgment.Strategy.NONE)
+    public CompletionStage<AqMessage<String>> betweenQueues(AqMessage<String> msg) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                PreparedStatement statement = msg.getDBConnection()
+                        .prepareStatement("INSERT INTO frank.message_log (message) VALUES (?)");
+                statement.setString(1, msg.getPayload());
+                statement.executeUpdate();
+            } catch (SQLException e) {
+                //Gets caught by messaging engine and translated to onError signal
+                throw new RuntimeException("Error when saving message to log table.", e);
+            }
+            return msg;
+        });
+    }
```

Comment:
@paulparkinson before starting the example build and run pre-populated Oracle DB
```bash
cd ./examples/messaging/docker/oracle-aq-18-xe/ && \
./buildAndRun.sh
```
Expect it to take tens of minutes before ready, 18.4 xe takes a long time to start